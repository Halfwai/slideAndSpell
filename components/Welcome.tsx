import { View, TouchableOpacity, StyleSheet, Animated } from 'react-native'
import { COLOURS } from '@/constants/colours'
import MyAppText from '@/components/MyAppText'
import AuthButton from '@/components/AuthButton'

import { useRef, useEffect } from 'react'

import { useRouter } from 'expo-router';

interface WelcomeProps {
    position: {x: number, y: number},
    setMenu: Function,
}

export default function Welcome(props: WelcomeProps) {
    const position = useRef(new Animated.ValueXY({x: 0, y: 0})).current;
    const router = useRouter();


    useEffect(() => {
        if(props.position.y === 0) {
            setTimeout(() => {
                slide()
            }, 300)
            return;
        }
        slide();   
    }, [props.position]);

    function slide(){
        Animated.timing(position, {
            toValue: { x: props.position.x, y: props.position.y },
            duration: 500,
            useNativeDriver: true,
        }).start();
    }

    return(
        <Animated.View style={{ transform: [{ translateX: position.x }, { translateY: position.y }], position: "absolute", width: '100%', alignItems: 'center' }}>
            <View style={styles.welcomeContainer}>
                <MyAppText style={styles.title}>Hi</MyAppText>
                <MyAppText style={styles.welcomeText}>Welcome to Slide and Spell, please sign in for stats and leaderboards, or hit endless mode to play without stats or tracking.</MyAppText>
            </View>
            <View style={styles.buttonContainer}>
                <AuthButton text="Sign In" onPress={() => props.setMenu("signIn")} style={{ backgroundColor: COLOURS.green}} />
                <AuthButton text="Sign Up" onPress={() => props.setMenu("signUp")} style={{ backgroundColor: "white", borderColor: COLOURS.green}} />
            </View>
            <View style={styles.endlessContainer}>
                <AuthButton text="Endless Mode" onPress={() => router.push('/endlessGame')} style={{ backgroundColor: "white", borderColor: COLOURS.green}} />            
            </View>

        </Animated.View>

    )
}

const styles = StyleSheet.create({
    welcomeContainer: {
        padding: 12,
        borderRadius: 10,
        width: '90%',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
        color: 'black',
        fontWeight: 'bold',
    },
    welcomeText: {
        fontSize: 15,
        color: 'black',
        marginBottom: 20,
        fontWeight: 'bold',
        textAlign: 'center',
    },
    buttonContainer: {
        flexDirection: 'column',
        justifyContent: 'space-between',
        width: '100%',
        alignItems: 'center',
    },
    button: {
        padding: 12,
        borderRadius: 10,
        marginBottom: 20,
        alignItems: 'center',
        borderWidth: 1,
        width: '90%',
    }, 
    buttonText: {
        fontSize: 20,
        color: 'black',
        fontWeight: 'bold',
    },
    endlessContainer: {
        height: "60%",
        width: "100%",
        alignItems: 'center',
        justifyContent: 'flex-end',
    }
})