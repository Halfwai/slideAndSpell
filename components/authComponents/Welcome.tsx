import React from 'react'
import { View, StyleSheet } from 'react-native'
import { useRouter } from 'expo-router';

// Import COLOURS
import { COLOURS } from '@/constants/colours'

// Import Components
import MyAppText from '@/components/common/MyAppText'
import AuthButton from '@/components/buttons/AuthButton'

// Setup props
interface WelcomeProps {
    setMenu: Function
}

export default function Welcome(props: WelcomeProps) {
    // get router
    const router = useRouter();
    return (
        <>
            <View style={styles.welcomeContainer}>
                <MyAppText style={styles.title}>Hi</MyAppText>
                <MyAppText style={styles.welcomeText}>Welcome to Slide and Spell, please sign in for stats and leaderboards, or hit endless mode to play without stats or tracking.</MyAppText>
            </View>
            <View style={styles.buttonContainer}>
                <AuthButton text="Sign In" onPress={() => props.setMenu("signIn")} style={{ backgroundColor: COLOURS.green }} />
                <AuthButton text="Sign Up" onPress={() => props.setMenu("signUp")} style={{ backgroundColor: "white", borderColor: COLOURS.green }} />
            </View>
            <View style={styles.endlessContainer}>
                <AuthButton text="Endless Mode" onPress={() => router.push('/endlessGame')} style={{ backgroundColor: "white", borderColor: COLOURS.green }} />
            </View>
        </>
    )
}

const styles = StyleSheet.create({
    welcomeContainer: {
        padding: 12,
        borderRadius: 10,
        width: '90%',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
        color: 'black',
        fontWeight: 'bold',
    },
    welcomeText: {
        fontSize: 15,
        color: 'black',
        marginBottom: 20,
        fontWeight: 'bold',
        textAlign: 'center',
    },
    buttonContainer: {
        flexDirection: 'column',
        justifyContent: 'space-between',
        width: '100%',
        alignItems: 'center',
    },
    button: {
        padding: 12,
        borderRadius: 10,
        marginBottom: 20,
        alignItems: 'center',
        borderWidth: 1,
        width: '90%',
    },
    buttonText: {
        fontSize: 20,
        color: 'black',
        fontWeight: 'bold',
    },
    endlessContainer: {
        height: "60%",
        width: "100%",
        alignItems: 'center',
        justifyContent: 'flex-end',
    }
})