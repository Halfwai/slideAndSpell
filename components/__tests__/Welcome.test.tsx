import { render } from '@testing-library/react-native';
import Welcome from '@/components/authComponents/Welcome';

it(`Welcome Component renders correctly`, () => {
    const { toJSON } = render(
        <Welcome
            setMenu={() => {}}
        />
    );    
    expect(toJSON()).toMatchSnapshot();
});