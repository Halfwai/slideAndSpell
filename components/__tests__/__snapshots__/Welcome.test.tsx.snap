// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component renders correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 10,
        "padding": 12,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Hi
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Welcome to Slide and Spell, please sign in for stats and leaderboards, or hit endless mode to play without stats or tracking.
    </Text>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Sign In
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Sign Up
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "height": "60%",
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Endless Mode
      </Text>
    </View>
  </View>,
]
`;
